### ISSUE TITLE
Inconsistent type inference v/s the type inferred by the compiler

### ISSUE THREAD

**Issue by liquidscorpio** [intellij-rust/intellij-rust#4351]
## Environment

* **IntelliJ Rust plugin version:**
0.2.104.2132-192

* **Rust toolchain version:**
rustc 1.36.0 (a53f9df32 2019-07-03)

* **IDE name and version:**
PyCharm 2019.2.1 (Professional Edition)
Build #PY-192.6262.63, built on August 22, 2019

* **Operating system:**
Ubuntu Bionic (18.04)

## Problem description
The code in the playground<sup>1</sup> fails in the compiler because it infers the type of `data` variable as `{integer}` and because of this limitation the compilation fails when methods specific to `i32|16` etc. like `abs()` are called. 

The plugin, however, infers it as `i32` (see red-labeled inference in the screenshot) and does not flag method calls like `abs()` on it.

<sup>1</sup> https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=41787bb7e7b18e4d7567c35d05ef9477

![sshot](https://user-images.githubusercontent.com/4113784/64258171-61614080-cf44-11e9-892b-b68f3e26ff56.png)

**PS**: To get the compiler to pass, one needs to edit the code and explicitly specify type for `data`. Like:
```rust
let data: [[i32; 2]; 5] = [.....]
```

## Steps to reproduce
Reproducible code sample in the attached playground.

## Desired outcome
The type inference on `data` should match with that of compiler and subsequent functional calls to `abs()` should be flagged as errors.
